### DISCUSSION TITLE
Handling OAuth Step up/challenge with AuthExchange

### DISCUSSION THREAD

**Discussion by crmitchelmore** [urql-graphql/urql#2709] [Q&A]
When we make a request the API can respond with an auth error along with [WWW-Authenticate](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/WWW-Authenticate) data, in REST this would be in headers but there could be a mapping to GQL errors. 

Is there any supported way to implement this with Auth Exchange? 

At the moment, I can't see it. I would respond `true` to `didAuthError` which would trigger `getAuth` but, as I see it, there's no way to get the error state to know what level of scope I need to request from the auth service.

**Comment by JoviDeCroock** [comment]
I think the best solution imho is to leverage that on the client this is a one-time process, this means you can do the following

```js
let scopeRequired;
authExchange({
  didAuthError: () => { scopeRequired = 'x'; return true; }
  getAuth: () => {
    let scope = scopeRequired;
    scopeRequired = undefined
  }
})
```
